tests/ert/unit_tests/scheduler/test_generic_driver.py

```diff
@@ -159,28 +159,45 @@ async def test_repeated_submit_same_iens(driver: Driver, tmp_path, monkeypatch):
 @pytest.mark.flaky(reruns=5)
 async def test_kill_actually_kills(driver: Driver, tmp_path, pytestconfig, monkeypatch):
     monkeypatch.chdir(tmp_path)
-    finished = False
+    realizations = {0, 1}
+    iens_status = {
+        realization: {"finished": False, "started": False}
+        for realization in realizations
+    }
+
     driver._poll_period = 0.01
 
-    async def kill_job_once_started(iens):
-        nonlocal driver
-        await driver.kill(iens)
+    async def kill_jobs_once_all_started(realizations):
```

Comment:
This callback is called for each of the realizations when they are started AFAIK, so you are not guaranteed that both are started when you get here. Check the definition of `poll() ` in `tests/ert/utils.py`